src/main/java/org/betonquest/betonquest/quest/condition/advancement/AdvancementCondition.java

```diff
@@ -0,0 +1,32 @@
+package org.betonquest.betonquest.quest.condition.advancement;
+
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.api.quest.condition.PlayerCondition;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.bukkit.advancement.Advancement;
+import org.bukkit.advancement.AdvancementProgress;
+
+/**
+ * Checks if the player has specified advancement.
+ */
+public class AdvancementCondition implements PlayerCondition {
+    /**
+     * Advancement which is required.
+     */
+    private final Advancement advancement;
+
+    /**
+     * Create a new Advancement condition.
+     *
+     * @param advancement the required advancement
+     */
+    public AdvancementCondition(final Advancement advancement) {
+        this.advancement = advancement;
+    }
+
+    @Override
+    public boolean check(final Profile profile) throws QuestRuntimeException {
+        final AdvancementProgress progress = profile.getOnlineProfile().get().getPlayer().getAdvancementProgress(advancement);
+        return progress.isDone();
+    }
```

Comment:
Can we find a design, that sole the issue mentioned by qodana that we need to call getOnlineProfile? may this is a n thing for an other PR

Reply:
For that we could expand the stuff with a sub type of PlayerCondition which requires an OnlineProfile.
When we do that for everything in the new API too we can just centralize that. The more I think about that the better it looks to me. We can get rid of these masses `profile.getOnlineProfile().get().getPlayer()` and have that in the Wrapper.

Reply:
Will be solved in a later PR